integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/utils/PublisherUtils.java

```diff
@@ -109,6 +114,33 @@ public static String createAPIUsingOAS(JSONObject apiProperties,
         return apiDTO.getId();
     }
 
+    /**
+     * Updates the OpenAPI definition of an already created API
+     * 
+     * @param apiId ID of the API to update the OpenAPI definition
+     * @param openAPIFileName Name of the OpenAPI file. ex. scopes_openAPI.yaml
+     * @param publisherRestClient An instance of RestAPIPublisherImpl
+     * @return ID of the API that was updated
+     * @throws CCTestException if the OpenAPI file specified was not present, or an error occurs when updating the API
+     */
+    public static String updateOpenAPIDefinition(String apiId, String openAPIFileName,
+                                               RestAPIPublisherImpl publisherRestClient) throws CCTestException {
+        String targetDir = Utils.getTargetDirPath();
+        Path definitionPath = Paths.get(targetDir + ApictlUtils.OPENAPIS_PATH + openAPIFileName);
+        String responseApiId;
+        try {
+            String openAPIContent = Files.readString(definitionPath);
+            responseApiId = publisherRestClient.updateSwagger(apiId, openAPIContent);
+        } catch (ApiException e) {
+            log.error("Error occurred while creating API with an OpenAPI definition. Response: {}", e.getResponseBody());
+            throw new CCTestException("Error while creating an API with an OpenAPI", e);
```

Comment:
Isn't it better to throw the detailed error here and log and handle the error in the catching method?

Reply:
When an error gets thrown during tests, it doesn't show the actual line the error occurred. Not sure why. Which is why we log the detail error then and there.